gatt/gatt_linux.py

```diff
@@ -208,6 +209,24 @@ def remove_device(self, mac_address):
         # TODO: Implement
         pass
 
+    def remove_all_devices(self, skip_alias=None):
+        self.update_devices()
+
+        keys_to_be_deleted = []
+        for key, device in self._devices.items():
+            if skip_alias and device.alias() == skip_alias:
+                continue
+            mac_address = device.mac_address.replace(':', '_').upper()
+            path = '/org/bluez/%s/dev_%s' % (self.adapter_name, mac_address)
+            self._adapter.RemoveDevice(path)
+            keys_to_be_deleted.append(key)
```

Comment:
You could do `del self._devices[key]` already here and get rid of `keys_to_be_deleted` state

Reply:
Apparently you can't iterate over a dict and mutate it at the same time in Python, hence this workaround.

```
>>> ll = {}
>>> ll = {"a": 1, "b": 2}
>>> ll
{'a': 1, 'b': 2}
>>> for k in ll:
...   del ll[k]
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: dictionary changed size during iteration
```

Reply:
Noted, will keep in mind for the next time. Thanks!